generate/generate.go

```diff
@@ -364,26 +360,40 @@ func newModDownloadRule(mod, version string, licences []string) (*build.CallExpr
 	return rule.Call, rule.Name()
 }
 
-func (u *Update) allSources(r *rule) ([]string, error) {
-	args := r.parseGlob()
-	if args == nil {
-		return r.AttrStrings("srcs"), nil
+func (u *Update) allSources(conf *config.Config, r *rule, sourceMap map[string]*GoFile) ([]string, map[string]*GoFile, error) {
```

Comment:
this function could do with a docstring to explain what the return values are

Reply:
Okay, done :)